specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json

```diff
@@ -173,15 +173,91 @@
           }
         },
         "x-ms-examples": {
           "WorkItemConfigurationDelete": {
             "$ref": "./examples/WorkItemConfigDelete.json"
           }
         }
+      },
+      "get": {
+        "description": "Gets specified work item configuration for an Application Insights component.",
+        "operationId": "WorkItemConfigurations_GetItem",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkItemConfigIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfiguration"
+              }
+            }
+          },
+          "x-ms-examples": {
+            "WorkItemConfigurationsGetDefault": {
+              "$ref": "./examples/WorkItemConfigGet.json"
+            }
+          }
+      },
+      "patch" : {
+        "description": "Update a work item configuration for an Application Insights component.",
+        "operationId": "WorkItemConfigurations_UpdateItem",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkItemConfigIdParameter"
+          },
+          {
+            "name": "WorkItemConfigurationProperties",
+            "description": "Properties that need to be specified to update a work item configuration for this Application Insights component.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/WorkItemCreateConfiguration"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful operation indicating the config updated by this PATCH call.",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsCreate": {
```

Comment:
Is this a long running operation

Reply:
It should be sub 20 seconds. But there is a chance it can run long.

Reply:
In that case this should be marked as long running operation

Reply:
What's the bar for that? I'm going to look at my metrics and see what the breakdown of this operation is.

Reply:
I have marked this for ARM review, they will help you understand it better.

Reply:
Thanks. I'll add this note here for reference when the team reviews

99th percentile is about 5.8 seconds
90th percentile is 1 second
75th percentile is 630ms
50th percentile is 303ms

Reply:
Based on the data, I think it's reasonable to keep it as a synchronous operation. However, you said "it can run long". Can you clarify what that means?

---
In reply to: [281306078](https://github.com/Azure/azure-rest-api-specs/pull/5870#discussion_r281306078) [](ancestors = 281306078)

Reply:
This is not a long running operation - it's just updating config in storage